pkg/operator/observe_config.go

```diff
@@ -179,6 +184,40 @@ func observeInternalRegistryHostname(listers Listers, observedConfig map[string]
 	return observedConfig, nil
 }
 
+func observeRoutingSubdomain(listers Listers, observedConfig map[string]interface{}) (map[string]interface{}, error) {
+	clusterConfig, err := listers.installConfigLister.ConfigMaps(clusterConfigNamespace).Get(clusterConfigName)
+	if errors.IsNotFound(err) {
+		return observedConfig, nil
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	if clusterConfig != nil {
+		if installConfig, ok := clusterConfig.Data["install-config"]; ok {
+			type InstallConfigMetadata struct {
+				Name string `json:"name"`
+			}
+
+			type InstallConfig struct {
+				Metadata   InstallConfigMetadata `json:"metadata"`
+				BaseDomain string                `json:"baseDomain"`
+			}
+
+			var config InstallConfig
+			if err := yaml.Unmarshal([]byte(installConfig), &config); err != nil {
+				return nil, fmt.Errorf("invalid InstallConfig: %v\njson:\n%s", err, installConfig)
+			}
+
+			subdomain := fmt.Sprintf("%s.%s", config.Metadata.Name, config.BaseDomain)
```

Comment:
Might as well add the `apps` prefix here until we sort out the cluster config?

Reply:
Done.  Thanks!